prov/sockets/src/sock_progress.c

```diff
@@ -2150,7 +2150,7 @@ static int sock_pe_progress_rx_pe_entry(struct sock_pe *pe,
 
 	if (sock_comm_is_disconnected(pe_entry)) {
 		SOCK_LOG_DBG("conn disconnected: removing fd from pollset\n");
-		if (pe_entry->conn->sock_fd != -1) {
+		if (pe_entry->ep_attr->cmap.used > 0) {
```

Comment:
When main thread acquires the progress lock first during fi_close(ep_fd), it sets all the fd = -1 to indicate it has been removed. So when progress thread resumes, the combination of map->used and fd value can determine if the entry was being freed by the main thread. I didn't add another flag to mark the fd invalid, instead set it to -1 to indicate that.